src/algebraic_topology/nerve.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebraic_topology.simplicial_set
+
+/-!
+
+# The nerve of a category
+
+This file provides the definition of the nerve of a category `C`,
+which is a simplicial set `nerve C` (see [goerss-jardine-2009], Example I.1.4).
+
+## References
+* [Paul G. Goerss, John F. Jardine, *Simplical Homotopy Theory*][goerss-jardine-2009]
+
+-/
+
+open category_theory.category
+
+universes v u
+
+namespace category_theory
+
+/-- The nerve of a category -/
+@[simps]
+def nerve (C : Type u) [category.{v} C] : sSet.{max u v} :=
+{ obj := λ Δ, (simplex_category.to_Cat.obj Δ.unop) ⥤ C,
+  map := λ Δ₁ Δ₂ f, ((whiskering_left _ _ _).obj (simplex_category.to_Cat.map f.unop)).obj,
+  map_id' := λ Δ, begin
+    erw functor.map_id,
```

Comment:
Hmm, I'm pretty worried here. My first step of diagnosing a mysterious `erw` is to try `dsimp` beforehand, and see what happens.

In this case, it is clear that we have too many simp lemmas, which are unfolding things into unpleasant forms. In particular, after `dsimp` at the beginning of this proof, we get
```
⊢ ((whiskering_left
        (NonemptyFinLinOrd_forget_to_Preorder.obj (NonemptyFinLinOrd.of (ulift (fin ((opposite.unop Δ).len + 1))))).α
        (NonemptyFinLinOrd_forget_to_Preorder.obj (NonemptyFinLinOrd.of (ulift (fin ((opposite.unop Δ).len + 1))))).α
        C).obj
       (simplex_category.to_Cat.map (simplex_category.hom.mk order_hom.id))).obj =
    𝟙 ((NonemptyFinLinOrd_forget_to_Preorder.obj
       (NonemptyFinLinOrd.of (ulift (fin ((opposite.unop Δ).len + 1))))).α ⥤ C)
```
This seems problematic on many fronts. I would hope not to see any of `.α`, `simplex_category.hom.mk`, or `ulift (fin ...)` in the simp normal form of this goal.

I would be inclined to attempt diagnosing and fixing (or deciding that they are actually sensible) these before proceeding. Might you be interested in looking into this (either as part of this PR or a preliminary one)? (It's okay if not.)

Reply:
I would finger guess that this due to overeager `simps`. I noticed it went crazy on functor applications for some reason, in particular when they are functors between `bundled` categories. We should probably tweak the custom simp projections.

Reply:
Actually, largely this is just that `@[simps]` is giving us over-aggressive lemmas for `to_Cat`. Could you try to get a better goal after `dsimp` (ideally so `rw functor.map_id` works), by instructing `@[simps]` to only give "more applied" lemmas, or just writing some of the simp lemmas for `to_Cat` by hand?

Reply:
My point is that the `to_Cat` situation that we see here actually happens all over the place.

Reply:
I am not sure how to handle this problem... Anyway, I believe I have at least circumvented it for now by cleaning up the definition and changing the proofs (by using `functor.comp_id` and `functor.id_comp`).

Reply:
Instead of tagging `to_Cat` with `simps`, tag it with `simps functor inverse`, and that will probably be enough for your purposes. You can check what lemmas it generates by writing `simps?`.

Reply:
As it is a functor, do you mean `@[simps obj map]` ?

Reply:
Done.